x/launch/types/message_request_add_vested_account.go

```diff
@@ -0,0 +1,57 @@
+package types
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+var _ sdk.Msg = &MsgRequestAddVestedAccount{}
+
+func NewMsgRequestAddVestedAccount(address, chainID string,
+	coins sdk.Coins, options *types.Any) *MsgRequestAddVestedAccount {
+	return &MsgRequestAddVestedAccount{
+		ChainID: chainID,
+		Address: address,
+		Coins:   coins,
+		Options: options,
+	}
+}
+
+func (msg *MsgRequestAddVestedAccount) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgRequestAddVestedAccount) Type() string {
+	return "RequestAddVestedAccount"
+}
+
+func (msg *MsgRequestAddVestedAccount) GetSigners() []sdk.AccAddress {
+	address, err := sdk.AccAddressFromBech32(msg.Address)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{address}
+}
+
+func (msg *MsgRequestAddVestedAccount) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgRequestAddVestedAccount) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Address)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid address (%s)", err)
+	}
+
+	_, _, err = ParseChainID(msg.ChainID)
+	if err != nil {
+		return sdkerrors.Wrap(ErrInvalidChainID, msg.ChainID)
+	}
+
+	if len(msg.Coins) == 0 {
+		return sdkerrors.Wrap(ErrEmptyCoins, msg.Address)
+	}
```

Comment:
We should also check that the provided `VestingOption` is one of the possible value for the any type (which now can only be `DelayedVesting`)

It is implemented for `InitialGenesis` for example: https://github.com/tendermint/spn/blob/c815fafa7096e7054027661e5677f659fe5e8298/x/launch/types/message_edit_chain.go#L67

I'm also thinking that we can check if the provided value is valid. For this we can define a method `validate() error` for the `VestingOption` interface and we call it here.
So we just verify that Coins is non empty and timestamp is not 0

I create an issue since we can do the same for `InitialGenesis`